src/Passes/Source/Logging/LogCollection.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Logging/ILogger.hpp"
+
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// Concrete ILogger implementation that quantum
+    class LogCollection : public ILogger
```

Comment:
I fail to find where the class `LogCollection` is used.
I fail to see where its member variables (`current_location_`, `messages_`) are read from.

Reply:
`LogCollection` is a work in progress that has not been added to the infrastructure. It is meant to go into the main QAT tool and passed to the passes through the interface to allow logging of errors and output these.